test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs

```diff
@@ -44,6 +45,10 @@ public HttpClientTests()
                         ctx.Response.RedirectLocation = "/";
                         ctx.Response.StatusCode = 302;
                     }
+                    else if (ctx.Request.Url.PathAndQuery.Contains("retry"))
```

Comment:
do we need any of this? the retry handler can simple make 'n' requests irrespective of the return code.